avocado/utils/network/interfaces.py

```diff
@@ -781,3 +784,65 @@ def validate_ipv4_netmask_format(self, netmask):
                 return False
             first_bit = False
         return True
+
+    def ping_flood(self, int_name, peer_ip, ping_count):
+        """
+        Function to start ping to remote machine with "-f" [ flood ] option,
+        on given interface.
+
+        Also this function enables to track the live data to determine the
+        ping flood failure, in case of failure the program will exit.
+
+        :param int_name: source interface name.
+        :param peer_ip: Peer IP address (IPv4 or IPv6)
+        :param ping_count: How many ICMP echo packets to send.
+        :return : returns True on sucessful ping flood.
+                  returns False on ping flood failure.
+        :rtype : boolean
+        """
+        cmd = f"ping -I {int_name} {peer_ip} -c {ping_count} -f "
```

Comment:
@naresh for generic ping there is already a "ping_check" module is available under "utils/network/interfaces"

This one is specific to "ping flood" because, in case of failure with 'f' option it takes lot of time to finish the command.

For ex :
ping 192.168.100.1 -c 10000 -f 
1. if the above command get success, it takes hardly 3 min to exit the command.
but when the same above command fails, it takes more than 10 min to exit command, where its un-necessary to wait until to command exit.

in this situation this module helps to capture live output of ping command with -f option and once it detects 10 packet failure it will exit the command immediately instead of waiting for 10 min.

and we cannot use the same module for both with "-f" and without "-f", because both command output is totally different from each other.
so my suggestion would be keep "ping_flood" as separate module.
or if you wish to any other pattern then we modify accordingly.

Reply:
ok